File: app/src/main/java/com/friendly_machines/frbpdoctor/watchprotocol/notification/WatchRawResponse.java
package com.friendly_machines.frbpdoctor.watchprotocol.notification;

import androidx.annotation.NonNull;

public class WatchRawResponse {
    int serial;
    int ackSerial;

    public short command;
    public byte[] arguments;

    public WatchRawResponse(int serial, int ackSerial, short command, byte[] arguments) {
        this.serial = serial;
        this.ackSerial = ackSerial;
        this.command = command;
        this.arguments = arguments;
    }

    @NonNull
    @Override
    public String toString() {
        return String.format("%d: %s", this.command, this.arguments);
    }
}

File: app/src/main/java/com/friendly_machines/frbpdoctor/watchprotocol/notification/WatchRawResponse.kt
package com.friendly_machines.frbpdoctor.watchprotocol.notification

class WatchRawResponse(val sequenceNumber: Int, val ackedSequenceNumber: Int, val command: Short, val arguments: ByteArray) {
    override fun toString(): String {
        return String.format("%d: %s", command, arguments)
    }
}